src/views/components/name_modal.vue

```diff
@@ -0,0 +1,105 @@
+<style>
+</style>
+
+<template>
+        <v-card>
+            <v-card-title>Edit Name</v-card-title>
+
+            <v-layout row wrap>
+                <v-flex xs12 sm6 md4>
+                        <v-menu
+                          ref="menu"
+                          :close-on-content-click="false"
+                          v-model="menu"
+                          :nudge-right="40"
+                          :return-value.sync="date"
+                          lazy
+                          transition="scale-transition"
+                          offset-y
+                          full-width
+                          min-width="290px">
+                        </v-menu>
+                </v-flex>
+
+                <v-flex xs12 sm6 md4>
+                    <v-text-field
+                        label="New Name"
+                        v-model="new_name"
+                        >
+                    </v-text-field>
+                </v-flex>
+            </v-layout>
+
+            <v-card-actions>
+                <v-btn
+                  color="error"
+                  flat
+                  @click="$emit('cancel')"
+                >
+                  Cancel
+                </v-btn>
+
+              <v-spacer></v-spacer>
+
+              <v-btn
+                color="primary"
+                flat
+                @click="saveName()">
+                Launch
+              </v-btn>
+
+            </v-card-actions>
+        </v-card>
+</template>
+
+<script>
+import generate_query_string from "./generate_query_string";
+export default {
+    name: "LaunchModal",
+    props: {
+        user_id: String
+    },
+    data() {
+        return {
+            date: null,
+            menu: false,
+            time: null,
+            new_name: null
+        };
+    },
+    methods: {
+        /**
+         * Function saves new name
+         */
+        saveName: async function() {
```

Comment:
No. Data handling doesn't happen in the widgets. It happpens in the main view `surveys.vue`. The widget passes back the edited name in an `$emit`